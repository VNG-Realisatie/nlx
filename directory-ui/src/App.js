// Copyright Â© VNG Realisatie 2020
// Licensed under the EUPL
//
import React from 'react'
import { Switch, BrowserRouter as Router, Route } from 'react-router-dom'
import { ThemeProvider } from 'styled-components'
import { GlobalStyles, DomainNavigation } from '@commonground/design-system'
import VersionLogger from './components/VersionLogger'
import Header from './components/Header'
import ServiceOverviewPage from './pages/ServicesOverviewPage'
import DocumentationPage from './pages/DocumentationPage'
import theme from './theme'
import '@fontsource/source-sans-pro/latin.css'

const App = () => (
  <>
    <ThemeProvider theme={theme}>
      <GlobalStyles />
      <Router>
        <DomainNavigation
          activeDomain="NLX"
          gitLabLink="https://gitlab.com/commonground/nlx/nlx"
        />

        <Header />

        <Switch>
          <Route
            path="/documentation/:organizationName/:serviceName"
            component={DocumentationPage}
          />
          <Route
            exact
            path="/:organizationName?/:serviceName?"
            component={ServiceOverviewPage}
          />
        </Switch>
      </Router>
      <VersionLogger />
    </ThemeProvider>
  </>
)

export default App
