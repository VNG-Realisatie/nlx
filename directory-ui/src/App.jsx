// Copyright Â© VNG Realisatie 2018
// Licensed under the EUPL

import React, { Fragment } from 'react'

import GlobalStyles from './components/GlobalStyles/GlobalStyles'
import Header from './components/Header/Header'
import ServiceOverviewPage from './pages/ServicesOverviewPage/ServicesOverviewPage'
import DocumentationPage from './pages/DocumentationPage/DocumentationPage'
import { VersionLogger } from '@commonground/design-system'

import { BrowserRouter as Router, Route } from 'react-router-dom'

const App = () => (
    <div className="App">
        <GlobalStyles/>
        <Router>
            <Fragment>
                <Header />

                <Route exact path="/" component={ServiceOverviewPage} />
                <Route
                    path="/documentation/:organizationName/:serviceName"
                    component={DocumentationPage}
                />
            </Fragment>
        </Router>
        <VersionLogger/>
    </div>
)

export default App
