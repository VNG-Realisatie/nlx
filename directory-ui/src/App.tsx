// Copyright Â© VNG Realisatie 2020
// Licensed under the EUPL
//
import React from 'react'
import { Routes, BrowserRouter, Route } from 'react-router-dom'
import { ThemeProvider } from 'styled-components'
import { GlobalStyles, DomainNavigation } from '@commonground/design-system'
import VersionLogger from './components/VersionLogger'
import Header from './components/Header'
import ServiceOverviewPage from './pages/services/ServicesPage'
import ParticipantsPage from './pages/participants'
import theme from './theme'
import '@fontsource/source-sans-pro/latin.css'

const App: React.FC = () => (
  <>
    <ThemeProvider theme={theme}>
      <GlobalStyles />
      <BrowserRouter>
        <DomainNavigation
          activeDomain="NLX"
          gitLabLink="https://gitlab.com/commonground/nlx/nlx"
        />

        <Header />

        <Routes>
          <Route path="/participants" element={<ParticipantsPage />} />
          <Route path="*" element={<ServiceOverviewPage />} />
        </Routes>
      </BrowserRouter>
      <VersionLogger />
    </ThemeProvider>
  </>
)

export default App
