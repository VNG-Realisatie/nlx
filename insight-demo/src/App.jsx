import React, { Fragment, Component } from "react"
import { Route, Link, Switch } from 'react-router-dom';
import posed, { PoseGroup } from 'react-pose';
import { media } from './theme/helpers'

import styled, { ThemeProvider } from 'styled-components'
import theme from './theme/themeConstants'
import GlobalStyle from './globalStyle'
import { Flex, Box } from '@rebass/grid'
import { Container } from './components/Grid/Grid'

import Button from "./components/Button/Button";
import Stepper from "./components/Stepper/Stepper";
import logo from './images/logo.svg'

// Pages
import Intro from './pages/Intro'
import StepOne from './pages/Step1'
import StepTwo from './pages/Step2'
import StepThree from './pages/Step3'
import StepFour from './pages/Step4'

const Header = styled.div`
    ${media.xsDown`
        padding-top: 32px;
        padding-bottom: 40px;
    `}

    ${media.xsUp`
        padding-top: 80px;
        padding-bottom: 32px;
    `}
`

const Logo = styled.div`
    ${media.xsDown`
        margin-bottom: 24px;
    `}

    ${media.xsUp`
        position: absolute;
        left: 0;
        right: 0;
        bottom: 1.75rem;
        margin: 0 auto;
    `}

    display: flex;
    justify-content: center;

    img {
        height: 16px;
    }
`

const RouteContainer = posed.div({
    before: { opacity: 0, x: ({ direction }) => direction === 'next' ? '100%' : '-100%',
    },
    enter: { opacity: 1, x: 0,
        transition: {
            default: {
                type: 'spring', stiffness: 30, mass: .5
            },
            opacity: {
                ease: 'easeInOut', duration: 750
            }
        }
    },
    exit: { opacity: 0, x: ({ direction }) => direction === 'next' ? '-100%' : '100%',
        transition: {
            default: {
                type: 'spring', stiffness: 30, mass: .5
            },
            opacity: {
                ease: 'easeInOut', duration: 300
            }
        }
    },
});

const StyledContainer = styled(Container)`
    position: relative;
    display: flex;
    flex-direction: column;
    min-height: 100vh;
    overflow: hidden;
`

const StyledPage = styled.div`
    flex-grow: 1;

    ${media.xsDown`
        padding-bottom: 6.5rem;
    `}

    ${media.xsUp`
        padding-bottom: 8rem;
        display: flex;
        align-items: center;
        justify-content: space-between;
    `}
`

const StyledLink = styled(Link)`
    text-decoration: none;
`

const StyledContent = styled.div`
    flex-shrink: 1;
    max-width: 584px;

    ${media.xsUp`
        padding: 0 2rem;
    `}
`

const LgButton = styled(Button)`
    z-index: 1;

    ${media.xsDown`
        display: none;
    `}

    ${media.xsUp`
        display: inline-flex;
    `}
`

const SmButton = styled(Button)`
    z-index: 1;
`

const Toolbar = styled(Flex)`
    position: fixed;
    bottom: 0;
    left: 0;
    right: 0%;
    padding: 12px 0 14px;
    background-color: white;
    box-shadow: rgba(0,0,0,0.03) 0px -1px 6px 0px, rgba(0,0,0,0.03) 0px -20px 20px -20px;

    ${media.xsUp`
        display: none;
    `}
`

class App extends Component {
    constructor(props) {
        super(props)

        this.state = {
            direction: 'next'
        }
    }

    setDirection = (direction) => {
        this.setState({direction})
    }

	render() {
		return (
			<ThemeProvider theme={theme}>
				<Route render={({ location }) => {
                    let prevLink
                    let nextLink
                    switch (location.pathname) {
                        case '/':
                            prevLink = null
                            nextLink = "/stepone"
                            break
                        case '/stepone':
                            prevLink = "/"
                            nextLink = "/steptwo"
                            break
                        case '/steptwo':
                            prevLink = "/stepone"
                            nextLink = "/stepthree"
                            break
                        case '/stepthree':
                            prevLink = "/steptwo"
                            nextLink = "/stepfour"
                            break
                        case '/stepfour':
                            prevLink = "/stepthree"
                            nextLink = null
                            break
                        default:
                            break
                    }

                    return (
                        <Fragment>
                            <GlobalStyle />
                            <StyledContainer>
                                <Header>
                                    <Container>
                                        <Flex alignItems="center" flexDirection="column">
                                            <Logo>
                                                <a href="https://nlx.io/" target="_blank" rel="noopener noreferrer">
                                                    <img src={logo} alt="logo" />
                                                </a>
                                            </Logo>
                                            <Stepper pathname={location.pathname} />
                                        </Flex>
                                    </Container>
                                </Header>
                                <StyledPage>
                                    {prevLink ?
                                        <StyledLink to={prevLink} onClick={() => this.setDirection('back')}>
                                            <LgButton variant="tertiary">Back</LgButton>
                                        </StyledLink>
                                        :
                                        <LgButton variant="tertiary" disabled >Back</LgButton>
                                    }
                                    <StyledContent>
                                        <PoseGroup preEnterPose="before" direction={this.state.direction}>
                                            <RouteContainer key={location.pathname}>
                                                <Switch location={location}>
                                                    <Route exact path="/" component={Intro} key="intro" />
                                                    <Route path="/stepone" component={StepOne} key="stepone" />
                                                    <Route path="/steptwo" component={StepTwo} key="steptwo" />
                                                    <Route path="/stepthree" component={StepThree} key="stepthree" />
                                                    <Route path="/stepfour" component={StepFour} key="stepfour" />
                                                </Switch>
                                            </RouteContainer>
                                        </PoseGroup>
                                    </StyledContent>
                                    {nextLink ?
                                        <StyledLink to={nextLink} onClick={() => this.setDirection('next')}>
                                            <LgButton>Next</LgButton>
                                        </StyledLink>
                                        :
                                        <LgButton disabled >Next</LgButton>
                                    }
                                    <Toolbar justifyContent="center" mt={6}>
                                        <Box mr={3}>
                                            {prevLink ?
                                                <StyledLink to={prevLink} onClick={() => this.setDirection('back')}>
                                                    <SmButton variant="tertiary">Back</SmButton>
                                                </StyledLink>
                                                :
                                                <SmButton variant="tertiary" disabled >Back</SmButton>
                                            }
                                        </Box>
                                        {nextLink ?
                                            <StyledLink to={nextLink} onClick={() => this.setDirection('next')}>
                                                <SmButton>Next</SmButton>
                                            </StyledLink>
                                            :
                                            <SmButton disabled >Next</SmButton>
                                        }
                                    </Toolbar>
                                </StyledPage>
                            </StyledContainer>
                        </Fragment>
                    )
                }} />
			</ThemeProvider>
		);
	}
}

export default App;
