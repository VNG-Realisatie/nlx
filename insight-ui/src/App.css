.App {
    display: flex;
}

.App__Sidebar {
    flex-basis: 200px;
    flex-shrink: 0;
    flex-grow: 0;
    background: #ffffff;
    border-right: 1px solid ;
}
