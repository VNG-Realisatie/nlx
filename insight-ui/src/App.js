import React, { Component } from 'react';
import { BrowserRouter } from 'react-router-dom';

import { withStyles, CssBaseline } from '@material-ui/core';
import { globalStyles} from './styles/muiTheme';

import ResponsiveDrawer from './components/Drawer';

class App extends Component {
    state={
        appTitle:"NLx Insights"
    }
    componentDidMount() {
        //this.props.getSessionData();
        //console.log(this.props)
    }

    render() {
        return (            
            <BrowserRouter>                
                <div className="App">
                    <CssBaseline />
                    <ResponsiveDrawer
                        appTitle={this.state.appTitle}/>
                </div>
            </BrowserRouter>            
        );
    }
}

export default withStyles(globalStyles)(App);
