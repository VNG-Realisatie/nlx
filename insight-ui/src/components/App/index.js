import React from 'react'
import { NavLink, Route } from 'react-router-dom'

import StyledApp, {StyledNLXNavbar, StyledContent} from './index.styles'
import GlobalStyles from '../../components/GlobalStyles'
import { Navigation } from '@commonground/design-system'
import SidebarContainer from '../../containers/SidebarContainer'

import HomePage from '../../components/HomePage'
import OrganizationPageContainer from '../../containers/OrganizationPageContainer'

const App = () =>
  <StyledApp>
    <GlobalStyles/>

    <StyledNLXNavbar homePageURL={process.env.REACT_APP_NAVBAR_HOME_PAGE_URL || 'https://www.nlx.io'}
                     aboutPageURL={process.env.REACT_APP_NAVBAR_ABOUT_PAGE_URL || 'https://www.nlx.io/about'}
                     docsPageURL={process.env.REACT_APP_NAVBAR_DOCS_PAGE_URL || 'https://docs.nlx.io'}>
      <Navigation.Item>
        <a href={process.env.REACT_APP_NAVBAR_DIRECTORY_URL}>Directory</a>
      </Navigation.Item>
      <Navigation.Item>
        <NavLink to="/">Insight</NavLink>
      </Navigation.Item>
    </StyledNLXNavbar>

    <SidebarContainer />

    <StyledContent>
      <Route path="/" exact component={HomePage} />
      <Route path="/organization/:organizationName/" component={OrganizationPageContainer} />
    </StyledContent>
  </StyledApp>

export default App
