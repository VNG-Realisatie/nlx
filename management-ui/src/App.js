// Copyright Â© VNG Realisatie 2020
// Licensed under the EUPL
//
import React, { Suspense } from 'react'
import { Switch, Route, Redirect } from 'react-router-dom'
import { ThemeProvider } from 'styled-components/macro'
import { GlobalStyles as DSGlobalStyles } from '@commonground/design-system'

import GlobalStyles from './components/GlobalStyles'
import theme from './theme'

import AddServicePage from './pages/AddServicePage'
import EditServicePage from './pages/EditServicePage'
import InwaysPage from './pages/InwaysPage'
import LoginPage from './pages/LoginPage/index'
import NotFoundPage from './pages/NotFoundPage'
import ServicesPage from './pages/ServicesPage'

import { StyledContainer } from './App.styles'

const App = () => (
  <StyledContainer>
    <ThemeProvider theme={theme}>
      <DSGlobalStyles />
      <GlobalStyles />

      {/* Suspense is required for XHR backend i18next */}
      <Suspense fallback={null}>
        <Switch>
          <Redirect exact path="/" to="/login" />
          <Route path="/login" component={LoginPage} />
          <Route path="/services/add-service" component={AddServicePage} />
          <Route
            path="/services/:name/edit-service"
            component={EditServicePage}
          />
          <Route path="/services" component={ServicesPage} />
          <Route path="/inways" component={InwaysPage} />

          <Route path="*" component={NotFoundPage} />
        </Switch>
      </Suspense>
    </ThemeProvider>
  </StyledContainer>
)

export default App
