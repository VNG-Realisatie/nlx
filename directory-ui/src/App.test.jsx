// Copyright Â© VNG Realisatie 2018
// Licensed under the EUPL

import React from 'react'
import { shallow } from 'enzyme'
import App from './App'

test('exists', () => {
  const wrapper = shallow(<App />)
  expect(wrapper.exists()).toBe(true)
})
