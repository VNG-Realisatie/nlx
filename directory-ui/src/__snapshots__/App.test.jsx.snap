// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<div
  className="App"
>
  <GlobalStyleComponent />
  <BrowserRouter>
    <Header />
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      path="/documentation/:organizationName/:serviceName"
    />
  </BrowserRouter>
  <VersionLogger
    logger={[Function]}
  />
</div>
`;
